<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1360</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Hearthstone Deck Choice Simulator</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTableWidget" name="resultsTable">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1311</width>
      <height>511</height>
     </rect>
    </property>
    <property name="editTriggers">
     <set>QAbstractItemView::NoEditTriggers</set>
    </property>
    <row>
     <property name="text">
      <string>1</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>2</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>3</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>4</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>5</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>6</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>7</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>8</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>9</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>10</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>11</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>12</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>13</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>14</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>15</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>16</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>17</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>18</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>19</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>20</string>
     </property>
    </row>
    <column>
     <property name="text">
      <string>Deck 1</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Deck 2</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Deck 3</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Tops</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Wins</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Losses</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Win Rates</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Number of Tops</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Percent Topped</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Occurrences</string>
     </property>
    </column>
   </widget>
   <widget class="QComboBox" name="biasDeck1">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>600</y>
      <width>130</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QComboBox" name="biasDeck2">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>600</y>
      <width>130</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QComboBox" name="biasDeck3">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>600</y>
      <width>130</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>580</y>
      <width>55</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Deck 1</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>580</y>
      <width>55</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Deck 2</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>580</y>
      <width>55</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Deck 3</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>580</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Composition Percent</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>580</y>
      <width>143</width>
      <height>80</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="SimulationProgressLabel">
         <property name="text">
          <string>No Simulations Running.</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Tournaments Run: </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="numTourneyLabel">
           <property name="text">
            <string>0</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPushButton" name="clearResultsButton">
       <property name="text">
        <string>Clear Results</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QCheckBox" name="scrubBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>670</y>
      <width>171</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Include Scrubs In Results</string>
    </property>
   </widget>
   <widget class="QComboBox" name="wrBox1">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>550</y>
      <width>130</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QComboBox" name="wrBox2">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>550</y>
      <width>130</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>530</y>
      <width>55</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Deck A</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>530</y>
      <width>55</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Deck B</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>530</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Win Rate of A vs B</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="wrSpinBox">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>550</y>
      <width>111</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <double>100.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>5.000000000000000</double>
    </property>
    <property name="value">
     <double>50.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="biasSpin">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>600</y>
      <width>121</width>
      <height>22</height>
     </rect>
    </property>
    <property name="maximum">
     <double>100.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>5.000000000000000</double>
    </property>
   </widget>
   <widget class="QLineEdit" name="biasLine">
    <property name="geometry">
     <rect>
      <x>718</x>
      <y>550</y>
      <width>111</width>
      <height>28</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>Add Player Name</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>520</y>
      <width>181</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Tournament Composition Bias</string>
    </property>
   </widget>
   <widget class="QPushButton" name="addBias">
    <property name="geometry">
     <rect>
      <x>649</x>
      <y>550</y>
      <width>71</width>
      <height>28</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="text">
     <string>Add</string>
    </property>
   </widget>
   <widget class="QPushButton" name="removeBias">
    <property name="geometry">
     <rect>
      <x>568</x>
      <y>550</y>
      <width>81</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>Remove</string>
    </property>
   </widget>
   <widget class="QComboBox" name="chooseBias">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>550</y>
      <width>248</width>
      <height>28</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>520</y>
      <width>243</width>
      <height>24</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Number of Tournaments:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="numSimulations">
       <property name="maximum">
        <number>999999999</number>
       </property>
       <property name="value">
        <number>100</number>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>550</y>
      <width>305</width>
      <height>24</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="label_11">
       <property name="text">
        <string>Number of Players per Tournament:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="numPlayers">
       <property name="maximum">
        <number>999999999</number>
       </property>
       <property name="value">
        <number>400</number>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>580</y>
      <width>129</width>
      <height>45</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QLabel" name="label_12">
         <property name="text">
          <string>Number of Rounds:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="roundsLabel">
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QLabel" name="label_14">
         <property name="text">
          <string>Top Cut:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="topCutLabel">
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="biasErrorLabel">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>520</y>
      <width>261</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="boLabel">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>630</y>
      <width>55</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Best of: </string>
    </property>
   </widget>
   <widget class="QLabel" name="bestOfLabel">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>630</y>
      <width>55</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>5</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>630</y>
      <width>511</width>
      <height>30</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_6">
     <item>
      <widget class="QPushButton" name="biasSave">
       <property name="text">
        <string>Save Bias</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="loadBias">
       <property name="text">
        <string>Load Bias</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="findBias">
       <property name="text">
        <string>Highlight Selected Bias</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>570</y>
      <width>261</width>
      <height>67</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <item>
      <widget class="QPushButton" name="defaultAllWR">
       <property name="text">
        <string>Default All Matchups</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_8">
       <item>
        <widget class="QPushButton" name="wrUpdate">
         <property name="text">
          <string>Update MU</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="wrDefault">
         <property name="text">
          <string>Default MU</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QProgressBar" name="progressBar">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>670</y>
      <width>1131</width>
      <height>24</height>
     </rect>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>570</y>
      <width>197</width>
      <height>30</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_9">
     <item>
      <widget class="QPushButton" name="saveButton">
       <property name="text">
        <string>Save Results</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="RunSimButton">
       <property name="text">
        <string>Run Simulation</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>630</y>
      <width>261</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_7">
     <item>
      <widget class="QPushButton" name="wrSave">
       <property name="text">
        <string>Save MUs</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="wrLoad">
       <property name="text">
        <string>Load MUs</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QCheckBox" name="elim1">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>610</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Single Elim.</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="elim2">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>640</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Double Elim.</string>
    </property>
   </widget>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>resultsTable</zorder>
   <zorder>biasDeck1</zorder>
   <zorder>biasDeck2</zorder>
   <zorder>biasDeck3</zorder>
   <zorder>label</zorder>
   <zorder>label_3</zorder>
   <zorder>label_4</zorder>
   <zorder>label_6</zorder>
   <zorder>scrubBox</zorder>
   <zorder>wrBox1</zorder>
   <zorder>wrBox2</zorder>
   <zorder>label_7</zorder>
   <zorder>label_8</zorder>
   <zorder>label_9</zorder>
   <zorder>wrSpinBox</zorder>
   <zorder>biasSpin</zorder>
   <zorder>biasLine</zorder>
   <zorder>label_10</zorder>
   <zorder>addBias</zorder>
   <zorder>removeBias</zorder>
   <zorder>chooseBias</zorder>
   <zorder>biasErrorLabel</zorder>
   <zorder>boLabel</zorder>
   <zorder>bestOfLabel</zorder>
   <zorder>progressBar</zorder>
   <zorder>elim1</zorder>
   <zorder>elim2</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1360</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
